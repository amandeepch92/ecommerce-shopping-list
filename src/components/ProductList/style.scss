
@import 'assets/scss/variables.scss';

/* Header */
.item-list {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    margin-bottom: 30px;
}