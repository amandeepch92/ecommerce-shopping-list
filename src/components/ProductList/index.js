import PropTypes from "prop-types";
import Product from "components/Product";
import "./style.scss";
/**
 * ProductList component can be used to display a listy of Products. This component is connected with Styled component
 *  to add style
 *  @param {object} items : items object contains array of object of prices, productNames and productType
 */
export default function ProductList(props) {
  const { items, className } = props;
  return (
    <ul className={className}>
      {items.map(item => (
        <Product className={'product-details'} item={item} key={item.index} />
      ))}
    </ul>
  );
};

ProductList.propTypes = {
  items: PropTypes.array.isRequired,
  className: PropTypes.string
};

