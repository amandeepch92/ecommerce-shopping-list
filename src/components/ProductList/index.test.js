import { shallow } from 'enzyme';
import ProductList from './index';
const { configure } = require("enzyme");
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });

const props = {
    productData: [
      {
        index: 0,
        isSale: true,
        isExclusive: false,
        price: "$18.88",
        productImage: "product-1.jpg",
        productName: "Striped shirt",
        size: ["S", "L", "XL"]
      },
      {
        index: 1,
        isSale: false,
        isExclusive: false,
        price: "$25.44",
        productImage: "product-2.jpg",
        productName: "Denim shirt",
        size: ["XS", "S"]
      }
    ]
  };
  
  describe("ProductList Component", () => {
    test("should render correctly", () => {
      const productlisting = shallow(<ProductList items={props.productData} />);
      expect(productlisting).toMatchSnapshot();
    });
  });
  