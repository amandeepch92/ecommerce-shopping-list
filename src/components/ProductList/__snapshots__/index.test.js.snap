// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component should render correctly 1`] = `
<ul>
  <Product
    className="product-details"
    item={
      Object {
        "index": 0,
        "isExclusive": false,
        "isSale": true,
        "price": "$18.88",
        "productImage": "product-1.jpg",
        "productName": "Striped shirt",
        "size": Array [
          "S",
          "L",
          "XL",
        ],
      }
    }
    key="0"
  />
  <Product
    className="product-details"
    item={
      Object {
        "index": 1,
        "isExclusive": false,
        "isSale": false,
        "price": "$25.44",
        "productImage": "product-2.jpg",
        "productName": "Denim shirt",
        "size": Array [
          "XS",
          "S",
        ],
      }
    }
    key="1"
  />
</ul>
`;
